src/libraries/System.DirectoryServices.Protocols/tests/VlvRequestControlTests.cs

```diff
@@ -64,9 +77,14 @@ public void Ctor_BeforeCount_AfterCount_StringTarget(int beforeCount, int afterC
             Assert.Equal(expectedValue, control.GetValue());
         }
 
+        public static IEnumerable<object[]> Ctor_BeforeCount_AfterCount_ByteArrayTarget_Data()
+        {
+            yield return new object[] { 0, 0, null, (PlatformDetection.IsWindows) ? new byte[] { 48, 132, 0, 0, 0, 18, 2, 1, 0, 2, 1, 0, 160, 132, 0, 0, 0, 6, 2, 1, 0, 2, 1, 0 } : new byte[] { 48, 14, 2, 1, 0, 2, 1, 0, 160, 6, 2, 1, 0, 2, 1, 0 } };
+            yield return new object[] { 10, 10, new byte[] { 1, 2, 3 }, (PlatformDetection.IsWindows) ? new byte[] { 48, 132, 0, 0, 0, 11, 2, 1, 10, 2, 1, 10, 129, 3, 1, 2, 3 } : new byte[] { 48, 11, 2, 1, 10, 2, 1, 10, 129, 3, 1, 2, 3 } };
```

Comment:
For long-term test-building maintenance, consider a helper.  E.g.

```C#
yield return new object[] { 0, 0, null, FormatLengths("30{0}020100020100A0{1}020100020100", 18, 6) };

...

private static byte[] FormatLengths(string hexFormat, params int[] lengths)
{
    string[] hexLengths = new string[lengths.Length];

    for (int i = 0; i < lengths.Length; i++)
    {
        int curLen = lengths[i];

        if (PlatformDetection.IsWindows)
        {
            hexLengths[i] = $"84{curLen:X8}";
        }
        else if (curLen < 0x80)
        {
            hexLengths[i] = curLen.ToString("X2");
        }
        else if (curLen <= byte.MaxValue)
        {
            hexLengths[i] = $"81{curLen:X2}";
        }
        else if (curLen <= ushort.MaxValue)
        {
            hexLengths[i] = $"82{curLen:X4}";
        }
        else if (curLen <= 0xFFFFFF)
        {
            hexLengths[i] = $"83{curLen:X6}";
        }
        else
        {
            hexLengths[i] = $"84{curLen:X8}";
        }
    }

    string hex = string.Format(hexFormat, hexLengths);
    // This presupposes something like the crypto tests have:
    return hex.HexToByteArray();
}
```

Reply:
I agree this would be valuable, but if that's ok with you I would rather do that in a subsequent PR so to not add more complexity to this PR for now. Once I have tests running on Mono and OSX and passing, then I think it would be a great time to address this.

Reply:
Deferring the test helper is totally fine.